### PR TITLE
synopsysctl gives failed return code #470

### PR DESCRIPTION
**Pull request by mphammer**
This could be very useful for scripting with synopsysctl to check if commands are successful.

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/488#" title="Author self-approved">mphammer</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [mphammer]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/synopsysctl/cmd_create.go [hunk 1]
```diff
@@ -350,7 +355,7 @@ func init() {
 	createAlertCtl.AddSpecFlags(createAlertCmd, true)
 	createCmd.AddCommand(createAlertCmd)
 
-	// Add Black Duck Command
+	// Add Black Duck Comman
```
Comment (msenmurugan) [inline pkg/synopsysctl/cmd_create.go]: ???

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/488#" title="Author self-approved">mphammer</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/488#pullrequestreview-252077798" title="Approved">msenmurugan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [mphammer,msenmurugan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/synopsysctl/cmd_create.go [hunk 2]
```diff
@@ -169,31 +176,28 @@ var createBlackDuckCmd = &cobra.Command{
 		log.Debugf("setting template spec %s", baseBlackDuckSpec)
 		err = createBlackDuckCtl.SwitchSpec(baseBlackDuckSpec)
 		if err != nil {
-			log.Errorf("%s", err)
-			return nil
+			cmd.Help()
+			return fmt.Errorf("%s", err)
```
Comment (rrati) [inline pkg/synopsysctl/cmd_create.go]: SwitchSpec should be returning an error type, so no need to fmt as errorf.  Just return the err

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/488#" title="Author self-approved">mphammer</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/488#pullrequestreview-252077798" title="Approved">msenmurugan</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/488#pullrequestreview-252289811" title="Approved">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [mphammer,msenmurugan,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
@rrati: once the present PR merges, I will cherry-pick it on top of release-2019.6.x in a new PR and assign it to you.

<details>

In response to [this](https://github.com/blackducksoftware/synopsys-operator/pull/488#issuecomment-504103254):

>/cherrypick release-2019.6.x


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/488#" title="Author self-approved">mphammer</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/488#pullrequestreview-252077798" title="Approved">msenmurugan</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/488#pullrequestreview-252504432" title="LGTM">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [mphammer,msenmurugan,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
@rrati: #488 failed to apply on top of branch "release-2019.6.x":
```Applying: synopsysctl gives failed return code #470
Using index info to reconstruct a base tree...
M	pkg/synopsysctl/cmd_create.go
M	pkg/synopsysctl/cmd_delete.go
M	pkg/synopsysctl/cmd_deploy.go
M	pkg/synopsysctl/cmd_describe.go
M	pkg/synopsysctl/cmd_edit.go
M	pkg/synopsysctl/cmd_get.go
M	pkg/synopsysctl/cmd_root.go
M	pkg/synopsysctl/cmd_start.go
M	pkg/synopsysctl/cmd_stop.go
M	pkg/synopsysctl/cmd_update.go
Falling back to patching base and 3-way merge...
Auto-merging pkg/synopsysctl/cmd_update.go
CONFLICT (content): Merge conflict in pkg/synopsysctl/cmd_update.go
Auto-merging pkg/synopsysctl/cmd_stop.go
CONFLICT (content): Merge conflict in pkg/synopsysctl/cmd_stop.go
Auto-merging pkg/synopsysctl/cmd_start.go
CONFLICT (content): Merge conflict in pkg/synopsysctl/cmd_start.go
Auto-merging pkg/synopsysctl/cmd_root.go
Auto-merging pkg/synopsysctl/cmd_get.go
CONFLICT (content): Merge conflict in pkg/synopsysctl/cmd_get.go
Auto-merging pkg/synopsysctl/cmd_edit.go
CONFLICT (content): Merge conflict in pkg/synopsysctl/cmd_edit.go
Auto-merging pkg/synopsysctl/cmd_describe.go
CONFLICT (content): Merge conflict in pkg/synopsysctl/cmd_describe.go
Auto-merging pkg/synopsysctl/cmd_deploy.go
Auto-merging pkg/synopsysctl/cmd_delete.go
CONFLICT (content): Merge conflict in pkg/synopsysctl/cmd_delete.go
Auto-merging pkg/synopsysctl/cmd_create.go
CONFLICT (content): Merge conflict in pkg/synopsysctl/cmd_create.go
error: Failed to merge in the changes.
Patch failed at 0001 synopsysctl gives failed return code #470

```

<details>

In response to [this](https://github.com/blackducksoftware/synopsys-operator/pull/488#issuecomment-504103254):

>/cherrypick release-2019.6.x


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Pull request merged**